typescript/src/vault/vault.ts

```diff
@@ -71,6 +73,40 @@ export class Vault {
         return new hookClass(hookState);
     }
 
+    /**
+     * Returns the max amount that can be swapped (in relation to the amount specified by user).
+     * @param maxSwapParams
+     * @returns Returned amount/scaling is respective to the tokenOut because that’s what we’re taking out of the pool and what limits the swap size.
+     */
+    getMaxSwapAmount(swapParams: MaxSwapParams, poolState: PoolState): bigint {
+        const pool = this.getPool(poolState);
+        return pool.getMaxSwapAmount(swapParams);
+    }
+
+    /**
+     * Returns the max amount of a single token that can be added to a pool.
+     * @param poolState
+     * @returns
+     */
+    getMaxSingleTokenAddAmount(poolState: PoolState): bigint {
+        const pool = this.getPool(poolState);
+        return pool.getMaxSingleTokenAddAmount();
+    }
+
+    /**
+     * Returns the max amount of a single token that can be removed from a pool.
+     * @param maxRemoveParams
+     * @param poolState
+     * @returns
+     */
+    getMaxSingleTokenRemoveAmount(
+        maxRemoveParams: MaxSingleTokenRemoveParams,
+        poolState: PoolState,
+    ): bigint {
+        const pool = this.getPool(poolState);
+        return pool.getMaxSingleTokenRemoveAmount(maxRemoveParams);
+    }
```

Comment:
Since most of `maxRemoveParams` are from `poolState`, we could simplify this external call by building maxRemoveParams in here:
```suggestion
    getMaxSingleTokenRemoveAmount(
        isExactIn: boolean,
        poolState: PoolState,
    ): bigint {
        const maxRemoveParams = {
           ...
        };
        const pool = this.getPool(poolState);
        return pool.getMaxSingleTokenRemoveAmount(maxRemoveParams);
    }
```
Same applies for `maxSwapParams`.
What do you think?

Reply:
Actually, I think there's not much benefit on doing that. Leave as is and I'll build inputs on my end